### PR TITLE
Implement zipAll (Issue #458)

### PR DESCRIPTION
**Pull request by adityavkk**
Implements `zipAll`

Unlike `zip`, `zipAll` continues zipping until the longest collection is exhausted. Missing values from shorter collections are filled with `undefined`.

This PR builds on some earlier work done in a previous [PR](https://github.com/facebook/immutable-js/pull/460)

### CODE REVIEW

File: src/Operations.js [hunk 1]
```diff
@@ -822,18 +825,22 @@ export function zipWithFactory(keyIter, zipper, iters) {
       let steps;
       if (!isDone) {
         steps = iterators.map(i => i.next());
-        isDone = steps.some(s => s.done);
+        if (zipAll) isDone = steps.every(s => s.done);
+        else isDone = steps.some(s => s.done);
       }
+
```
Comment (arjans) [inline src/Operations.js]: This is the critical change in logic, as @leebyron [pointed out](https://github.com/facebook/immutable-js/pull/460/files#r29642099) in an earlier pull request.

File: dist/immutable.js [hunk 2]
```diff
@@ -1848,18 +1851,22 @@ function zipWithFactory(keyIter, zipper, iters) {
...
+
       if (isDone) {
         return iteratorDone();
       }
+
       return iteratorValue(
         type,
         iterations++,
-        zipper.apply(null, steps.map(function (s) { return s.value; }))
+        zipper.apply(null, steps.map(function (s) { return s.done ? undefined : s.value; }))
```
Comment (leebyron) [inline dist/immutable.js]: When `s.done` is true, `s.value` is guaranteed to be `undefined`, so is this change necessary?

File: dist/immutable.js [hunk 3]
```diff
@@ -1848,18 +1851,22 @@ function zipWithFactory(keyIter, zipper, iters) {
       var steps;
       if (!isDone) {
         steps = iterators.map(function (i) { return i.next(); });
-        isDone = steps.some(function (s) { return s.done; });
+        if (zipAll) { isDone = steps.every(function (s) { return s.done; }); }
+        else { isDone = steps.some(function (s) { return s.done; }); }
```
Comment (leebyron) [inline dist/immutable.js]: Please use a ternary for this:

```
isDone = zipAll ? steps.every(...) : steps.some(...)
```
